tests/bson-deserialize-typewrapped-binary-001.phpt

```diff
@@ -0,0 +1,105 @@
+--TEST--
+BSON serialization: type wrapped Binary
+--FILE--
+<?php
+$tests = [
+	[ 'types' => [] ],
+	[ 'types' => [ 'Binary' => 'MyBinary' ] ],
+	[ 'root' => 'array', 'types' => [ 'Binary' => 'MyBinary' ] ],
+	[ 'types' => [ 'Binary' => 'MyBinaryReturnScalarType' ] ],
+];
+
+class MyBinary implements \MongoDB\BSON\TypeWrapper
```

Comment:
Doing that in the test cases mean I need to implement all the methods for that too. I don't think that's useful in the tests as it's only going to be extraneous boiler plate code. For user examples, they should be there of course.